mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSink.java

```diff
@@ -44,6 +47,16 @@ public SqlServerSink(SqlServerSinkConfig sqlServerSinkConfig) {
     this.sqlServerSinkConfig = sqlServerSinkConfig;
   }
 
+  @Override
+  protected SchemaReader getSchemaReader() {
+    return new SqlServerSchemaReader.Sink();
```

Comment:
what do we achieve with those changes in general? Why do we need additional "schema readers" if we already have DB specific Sink and Source implementations?

Reply:
DB specific Sink and Source implementations are based on common implementation, which uses `CommonSchemaReader`.
To customize handling of DB-specific types we must use specific implementations of the schema reader, such as:
 
* `AuroraPostgresSchemaReader`
* `DB2SchemaReader`
* `OracleSchemaReader`
* `PostgresSchemaReader`


For instance, `SqlServerSchemaReader` allows us to map `DATETIMEOFFSET` data type to `STRING` and `GEOMETRY`, `GEOGRAPHY` types to `BYTES`.